app/mqtt_reader.py

```diff
@@ -27,29 +11,40 @@
 
 from app.message import CommandMessage
 from app.exceptions import InvalidMessageError
+from app.error_handler import ErrorHandler
 
 
 def _decode_message(message):
     return message.payload.decode()
 
 
-class MqttClient:
-    _client: mqtt.Client
-
-    def __init__(self, port: int, host: str, client: mqtt.Client) -> None:
-        self._client = client
+class MqttReader:
+    def __init__(
+        self,
+        host: str,
+        port: int,
+        topics: list[str],
+        client: mqtt.Client,
+        error_handler: ErrorHandler,
+    ) -> None:
         self.host = host
         self.port = port
+        self.error_handler = error_handler
         self.on_message_callbacks = []
-        self.topics = []
-
-    def subscribe_topics(self, topics: list[str]):
-        for topic in topics:
-            self.topics.append(topic)
+        self.topics = topics
```

Comment:
Any reason to not have this as private?

Reply:
Hmmm. Making it private would indicate that it shouldn't be modified, but there might be a valid reason to be able to see the value. I'll make it private and we can add an accessor method later if needed